zeus/run/dataloader.py

```diff
@@ -104,40 +110,43 @@ class ZeusDataLoader(DataLoader):
     model = torchvision.models.resnet18()
     torch.cuda.set_device(local_rank)
     model.cuda(local_rank)
-    # Zeus only supports one process per GPU profiling. If you are doing data
+    # Zeus assumes that exactly one process manages one GPU. If you are doing data
     # parallel training, please use `DistributedDataParallel` for model replication
-    # and specify the `device_ids` and `output_device` correctly.
+    # and specify the `device_ids` and `output_device` as below:
     model = torch.nn.parallel.DistributedDataParallel(
         model,
         device_ids=[local_rank],
         output_device=local_rank,
     )
 
     # Step 3: Create instances of `DistributedSampler` to partition the dataset
-        # across the GPUs.
+    # across the GPUs.
     train_sampler = torch.utils.data.distributed.DistributedSampler(train_set)
     eval_sampler = torch.utils.data.distributed.DistributedSampler(eval_set)
 
-    # Step 4: Create instances of `ZeusDataLoader`.
-    # Pass `"dp"` to `distributed` and samplers in the previous step to
-    # `sampler`.
+    # Step 4: Instantiate `ZeusDataLoader`.
+    # `distributed="dp` tells `ZeusDataLoader` to operate in data parallel mode.
     # The one instantiated with `max_epochs` becomes the train dataloader.
     train_loader = ZeusDataLoader(train_set, batch_size=256, max_epochs=100,
-                                sampler=train_sampler, distributed="dp")
+                                  sampler=train_sampler, distributed="dp")
     eval_loader = ZeusDataLoader(eval_set, batch_size=256, sampler=eval_sampler,
-                                distributed="dp")
+                                 distributed="dp")
 
-    # Step 5: Put your training code here.
+    # Step 5: Training loop.
+    # Use the train dataloader's `epochs` generator to allow Zeus to early-stop
+    # based on the training cost. Use `report_metric` to let Zeus know the current
```

Comment:
```suggestion
    # based on the cost. Use `report_metric` to let Zeus know the current
```
I'm thinking the cost includes training cost as well the evaluation cost lol.

Reply:
"Training" cost kinda means the cost of the DNN training job, but yeah, let's consistently just say cost.